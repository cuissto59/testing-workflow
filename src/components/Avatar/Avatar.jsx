import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";


const avatarSize = {
    full: 'w-full', /* 100% */
    xs: 'w-8',  /* 32px */
    sm: 'w-10',  /* 40px */
    base: 'w-24', /* 96px */
    l: 'w-32', /* 128px */ 
    xl: 'w-40', /* 160px */
};

export const Avatar = ({size, alt, src}) => {
    return <img
        className={clsx(avatarSize[size], 'rounded-full overflow-hidden shadow-inner')}
        src={src}
        alt={alt}
    />

}

Avatar.propTypes = {
    size: PropTypes.oneOf(['xs', 'sm', 'base', 'l', 'xl', 'full']),
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
};

Avatar.defaultProps = {
    size: 'full',
}
