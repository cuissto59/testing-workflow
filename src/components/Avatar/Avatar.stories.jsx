import {Avatar} from "./Avatar";


export default {
    title: 'Avatar',
    component: Avatar,
    argsTypes: {
        size: {
            defaultValue: 'l',
            description: 'size desc',
            options: ['xs', 'sm', 'base', 'l', 'xl'],
            control: {type: 'select'},

        },
        src: {
            description: 'size desc',
            control: {type: 'text'}

        }
    }
}

const Template = (args) => <Avatar {...args}/>

export const VerySmall = Template.bind({})

VerySmall.args = {
    size: 'xs',
    src: 'https://pickaface.net/gallery/avatar/unr_random_180410_1905_z1exb.png',
    alt: 'avatar'
}
export const Small = Template.bind({})
Small.args = {
    size: 'sm',
    src: 'https://pickaface.net/gallery/avatar/unr_random_180410_1905_z1exb.png',
    alt: 'avatar'
}
export const Base = Template.bind({})

Base.args = {
    size: 'base',
    src: 'https://pickaface.net/gallery/avatar/unr_random_180410_1905_z1exb.png',
    alt: 'avatar'
}
export const Large = Template.bind({})

Large.args = {
    size: 'l',
    src: 'https://pickaface.net/gallery/avatar/unr_random_180410_1905_z1exb.png',
    alt: 'avatar'
}
export const VeryLarge = Template.bind({})
VeryLarge.args = {
    size: 'xl',
    src: 'https://pickaface.net/gallery/avatar/unr_random_180410_1905_z1exb.png',
    alt: 'avatar'
}
