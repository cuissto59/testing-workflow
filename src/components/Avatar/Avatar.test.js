import {render} from '@testing-library/react';
import {Avatar} from "./Avatar";
import React from "react";

describe("Avatar", () => {
  it("renders correctly", () => {
    // eslint-disable-next-line no-undef
    const {container} = render(React.createElement(Avatar, {
      src: "https://avatars2.githubusercontent.com/u/1234?s=460&v=4",
      alt: "Avatar"
    }));
    expect(container).toMatchSnapshot();
  });
});