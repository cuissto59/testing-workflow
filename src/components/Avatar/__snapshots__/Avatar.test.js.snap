// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div>
  <img
    alt="Avatar"
    class="w-full rounded-full overflow-hidden shadow-inner"
    src="https://avatars2.githubusercontent.com/u/1234?s=460&v=4"
  />
</div>
`;
